app/models/concerns/dirty_associations.rb

```diff
@@ -57,6 +57,10 @@ def dirty_has_many_associations(*args)
               value
             end
           end
+
+          define_method "#{association_ids}_change" do
+            [instance_variable_get("@#{association_ids}_was"), self.send(association_ids)]
```

Comment:
why not using `self.send("#{association_ids}_was")` for the first item of the array, it would reuse the cache and properly initialize it for new record